odincal/odincal_handler.py

```diff
@@ -0,0 +1,178 @@
+            ac_file,
+            ac_stw_start,
+            ac_stw_end,
+            buffer,
+        )
+        raise BadAttitude(msg)
+
+
+def notify_queue(
+    sqs_client,
+    notification_queue,
+    scans,
+):
+    response = sqs_client.send_message(
+        QueueUrl=notification_queue,
+        MessageBody=json.dumps({
+            "scans": scans,
+        }),
+        MessageGroupId="Level1bScans",
+    )
+    if response["ResponseMetadata"]["HTTPStatusCode"] != 200:
+        msg = "Notification failed for scans {0} with status {}".format(
+            scans,
+            response["ResponseMetadata"],
+        )
+        raise NotifyFailed(msg)
+
+
+def handler(event, context):
+    pg_host_ssm_name = get_env_or_raise("ODIN_PG_HOST_SSM_NAME")
+    pg_user_ssm_name = get_env_or_raise("ODIN_PG_USER_SSM_NAME")
+    pg_pass_ssm_name = get_env_or_raise("ODIN_PG_PASS_SSM_NAME")
+    pg_db_ssm_name = get_env_or_raise("ODIN_PG_DB_SSM_NAME")
+    psql_bucket = get_env_or_raise("ODIN_PSQL_BUCKET_NAME")
+    notification_queue = get_env_or_raise("ODIN_L1_NOTIFICATIONS")
+
+    with TemporaryDirectory(
+        "psql",
+        "/tmp",
+    ) as psql_dir:
+        s3_client = boto3.client('s3')
+
+        # Setup SSL for Postgres
+        pg_cert_path = download_file(
+            s3_client,
+            psql_bucket,
+            psql_dir,
+            "/postgresql.crt",
+        )
+        root_cert_path = download_file(
+            s3_client,
+            psql_bucket,
+            psql_dir,
+            "/root.crt",
+        )
+        pg_key_path = download_file(
+            s3_client,
+            psql_bucket,
+            psql_dir,
+            "/postgresql.key",
+        )
+        os.chmod(pg_key_path, stat.S_IWUSR | stat.S_IRUSR | stat.S_IRGRP)
+        os.environ["PGSSLCERT"] = pg_cert_path
+        os.environ["PGSSLROOTCERT"] = root_cert_path
+        os.environ["PGSSLKEY"] = pg_key_path
+        import psycopg2
+
+        ssm_client = boto3.client("ssm")
+        db_host = ssm_client.get_parameter(
+            Name=pg_host_ssm_name,
+            WithDecryption=True,
+        )["Parameter"]["Value"]
+        db_user = ssm_client.get_parameter(
+            Name=pg_user_ssm_name,
+            WithDecryption=True,
+        )["Parameter"]["Value"]
+        db_pass = ssm_client.get_parameter(
+            Name=pg_pass_ssm_name,
+            WithDecryption=True,
+        )["Parameter"]["Value"]
+        db_name = ssm_client.get_parameter(
+            Name=pg_db_ssm_name,
+            WithDecryption=True,
+        )["Parameter"]["Value"]
+        
+        pg_string = "host={0} user={1} password={2} dbname={3} ssl-mode=verify-ca".format(  # noqa: E501
+            db_host,
+            db_user,
+            db_pass,
+            db_name,
+        )
+        with psycopg2.connect(pg_string) as con:
+            ac_file, backend, version = parse_event_message(event)
+            assert_has_attitude_coverage(ac_file, backend, version, con)
```

Comment:
A bit curious, what happens with the message when the lambda fails?

Reply:
Just to document what I said at stand-up: 
* The lambda is set up to subscribe to a queue with a number N of retries, a visibility time-out and a dead letter queue.
* When a message fails the first N times it will be made invisible for the set amount of time (_e.g._ 24 h), then it will be put back in the original queue, and the lambda will then try to process it again.
* After the retries have been exhausted the message will be put in the DLQ, from which we have the option to "redrive" it after any issues have been fixed.
* The message remains in the DLQ for 14 days, unless it is redriven, after which the 14 day timer restarts. 

This is is the workflow we use in MATS. I think a chain of step functions might be a better option (the first step would be checking coverage, which is currently done as the first step in the handler), but the method as it stands should work for Odin as well.